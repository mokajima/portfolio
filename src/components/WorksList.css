.section {
  margin: 0 auto;
  max-width: 1450px;
  width: 90%;
}

.headline {
  font: 400 2em "Pacifico", cursive, sans-serif;
  text-align: center;
  margin-bottom: 0.5em;
}

.featured-work {
  display: grid;
  grid-template-columns: 1fr;
  grid-row-gap: 30px;
  margin-bottom: 30px;
}

@media screen and (min-width: 500px) {
  .headline {
    font-size: 3.125em;
    margin-bottom: 25px;
  }
}

@media screen and (min-width: 600px) {
  .featured-work {
    grid-template-columns: 1fr 1fr;
    grid-column-gap: 5%;
  }
}

@media screen and (min-width: 992px) {
  .section {
    margin-bottom: 100px;
  }

  .featured-work {
    grid-template-columns: 1fr 1fr 1fr;
  }
}
