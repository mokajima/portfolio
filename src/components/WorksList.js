import React from 'react'
import Work from './Work'

function WorksList(props) {
  return (
    <section className="section">
      <h2 className="headline">Featured Work</h2>
      <ul className="featured-work">
        {props.works.map((work) => (
          <li key={work.id}>
            <Work work={work} />
          </li>
        ))}
      </ul>
    </section>
  )
}

export default WorksList
