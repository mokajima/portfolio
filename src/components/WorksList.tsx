import React, { FC } from 'react'
import Work from './Work'
import './WorksList.css'

interface Props {
  works: Work[];
}

const WorksList: FC<Props> = ({
  works
}) => (
  <section className="section">
    <h2 className="headline">Featured Work</h2>
    <ul className="featured-work">
      {works.map((work: Work) => (
        <li key={work.id}>
          <Work work={work} />
        </li>
      ))}
    </ul>
  </section>
)

export default WorksList
