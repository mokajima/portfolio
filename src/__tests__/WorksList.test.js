import React from 'react'
import WorksList from '../components/WorksList'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const works = [{
    id: 'readable',
    title: 'Readable',
    summary: 'An anonymous content and comment web application.',
    url: 'https://github.com/mokajima/readable',
    border: true,
    date: '2018-12'
  }]

  const tree = renderer
    .create(<WorksList works={works} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
