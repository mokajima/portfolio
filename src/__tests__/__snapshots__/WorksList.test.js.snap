// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="section"
>
  <h2
    className="headline"
  >
    Featured Work
  </h2>
  <ul
    className="featured-work"
  >
    <li>
      <div
        className="work"
      >
        <a
          href="https://github.com/mokajima/readable"
          rel="noopener noreferrer"
          target="_blank"
        >
          <figure
            className="work__img work__img--border"
          >
            <img
              alt="Readable"
              src="images/readable.jpg"
            />
          </figure>
        </a>
        <h3
          className="work__title"
        >
          <a
            href="https://github.com/mokajima/readable"
            rel="noopener noreferrer"
            target="_blank"
          >
            Readable
          </a>
        </h3>
        <p
          className="work__summary"
        >
          An anonymous content and comment web application.
        </p>
      </div>
    </li>
  </ul>
</section>
`;
